hazelcast/src/test/java/com/hazelcast/internal/serialization/impl/compact/GenericRecordTest.java

```diff
@@ -234,4 +234,38 @@ private void assertSetterThrows(GenericRecordBuilder builder, String fieldName,
                 .isInstanceOf(HazelcastSerializationException.class)
                 .hasMessageStartingWith(errorMessage);
     }
+
+    @Test
+    public void testGetFieldThrowsExceptionWhenFieldDoesNotExist() throws IOException {
+        GenericRecord record = compact("test").build();
+        assertThatThrownBy(() -> {
+            record.getInt32("doesNotExist");
+        }).isInstanceOf(HazelcastSerializationException.class).hasMessageContaining("Invalid field name");
+
+        InternalSerializationService serializationService = (InternalSerializationService) createSerializationService();
+        Data data = serializationService.toData(record);
+
+        InternalGenericRecord internalGenericRecord = serializationService.readAsInternalGenericRecord(data);
+
+        assertThatThrownBy(() -> {
+            internalGenericRecord.getInt32("doesNotExist");
+        }).isInstanceOf(HazelcastSerializationException.class).hasMessageContaining("Unknown field name");
```

Comment:
Minor: why do we have different error messages for the same thing with different generic records?

Reply:
I don't know, looks like it ended up this way. Let me try to reunite them.